yt/frontends/chimera/tests/test_outputs.py

```diff
@@ -0,0 +1,195 @@
+"""
+Chimera frontend tests
+
+"""
+
+import numpy as np
+
+from yt.testing import assert_almost_equal, assert_array_equal, assert_equal
+from yt.utilities.answer_testing.framework import (
+    GenericArrayTest,
+    data_dir_load,
+    requires_ds,
+)
+
+Two_D = "F37_80/chimera_00001_grid_1_01.h5"
+
+
+@requires_ds(Two_D)
+def test_2D():
+    ds = data_dir_load(Two_D)
+    _fields = [
+        ("chimera", "    n"),
+        ("chimera", "    p"),
+        ("chimera", "  c12"),
+        ("chimera", "  he4"),
+        ("chimera", "  o16"),
+        ("chimera", "  s32"),
+        ("chimera", " ar36"),
+        ("chimera", " ca40"),
+        ("chimera", " cr48"),
+        ("chimera", " fe52"),
+        ("chimera", " fe56"),
+        ("chimera", " mg24"),
+        ("chimera", " ne20"),
+        ("chimera", " ni56"),
+        ("chimera", " si28"),
+        ("chimera", " ti44"),
+        ("chimera", " zn60"),
```

Comment:
Why do these field names contain leading whitespace ? I think these ought to be trimmed off (both in this test and wherever the names are read in the frontend)

Reply:
If these are from the file (which they likely are, since they're `chimera`-type) they likely have that whitespace padding in the file.  This often shows up when reading from fixed-size string arrays, like `|S4` or something, in hdf5.

Reply:
I get that, but is there a reason for keeping that whitespace as part of the API ?

Reply:
We aren't leaving them as part of the API; this is reflecting the exact field names on disk.  When they get aliased (if they get aliased) they won't be `chimera` field-type anymore.

Reply:
Ah, I see that we're actually having a divergence of opinions. I consider even "raw" fields to be part of the public API because
1) they are de-facto accessible from the user side (and without any additional hurdle, as compared to aliased/derived fields)
2) in the absence of an alias they may be the _only_ way to access some fields

If that's not convincing to you I'll withdraw my suggestion.

Reply:
I suppose my thought process is that we shouldn't necessarily break the idea of iterating over a list that comes from somewhere else, that might have those specific field names from the file.  I am not unconvinced yet; @Ronan-Hix what do you think?

Reply:
These do indeed come from disk. They're fixed-length name arrays that come from Fortan and get stored that way in the hdf5 file. I could, I suppose, have it trim the whitespace off the names it reads in when generating the fields list, then add a couple of lines in the io routine that would add them back to match the data path, although a hardcoded solution like that might open up compatibility issues if something changes on the data storage side.

Reply:
Sorry I may be a little slow here but I don't get why you'd want to add it back ?

Reply:
I'm not necessarily against making that change, my only aversion to adding the whitespace back in the io system is that it might break if the chimera file system ever changes (For instance if the whitespace gets removed from the names on disk), whereas the purely procedural system won't.

Reply:
I think what I'm actually proposing is to map on-disk names to their trimmed equivalent at runtime _and keep a reference to that mapping in the indexing process_ so we can consistently access fields without having to readd any amount of whitespace at any point. I think this is actually pretty resilient to upstream changes. I did something similar in another frontend where on-disk names are sometimes upper case and some other times lower case, but I normalize them to lower case.

Reply:
Which frontend is that implementation in? I can use it as a reference to add whitespace trimming.

Reply:
that's in an externally maintained frontend for idefix/Pluto

here's the relevant part

https://github.com/neutrinoceros/yt_idefix/blob/3fcf1a01d0a73eb3539eb242e96f64226bca49ba/yt_idefix/_io/vtk_io.py#L247-L250